File: e2e/playwright-typescript/interactions/validators/AccessibilityValidator.ts
import { expect, test } from '@playwright/test';
import type { AxeResults } from 'axe-core';
import { step } from '@/decorators/step-decorator';

export class AccessibilityValidator {
  @step('Validate no accessibility violations')
  async hasNoViolations(results: AxeResults): Promise<void> {
    await test.info().attach('Axe accessibility report', {
      body: JSON.stringify(results.violations, null, 2),
      contentType: 'application/json',
    });
    expect(results.violations, 'Expected no accessibility violations').toEqual([]);
  }
}

File: e2e/capybara-ruby/interactions/validators/accessibility_validator.rb
# frozen_string_literal: true

require 'json'

require_relative '../../decorators/step_decorator'
require_relative '../../support/capybara_helpers'

class AccessibilityValidator
  extend StepDecorator::Methods
  include RSpec::Matchers

  step 'Validate no accessibility violations'
  def has_no_violations(results)
    Allure.add_attachment(
      name: 'Axe accessibility report',
      source: JSON.pretty_generate(results['violations']),
      type: Allure::ContentType::JSON
    )
    expect(results['violations']).to eq([])
  end
end
